### DISCUSSION TITLE
Questions about viewing traces

### DISCUSSION THREAD

**Discussion by alexpdp7** [grafana/clickhouse-datasource#711] [Q&A] [answered]
So I was testing the new and updated Clickhouse integration in Grafana for traces. I love how much it works OOB (just select the query builder, switch the query type to traces, and I can run a query, click on a trace and view it.

However, I'm struggling a bit with a few things:

* Can I add columns to the query? I just get the traceID, serviceName, operationName, startTime, and duration. I'd like to be able to select some span attribute, status code/messages, events... Of course viewing thing information in a table is not *that* useful, but it would help with filtering (e.g. viewing the status code so I can filter "ok" traces quickly).
* Similarly, when I click through to a trace, I miss being able to see the status and the events; when using the Python SDK, exceptions get recorded in the status and into an event, but I cannot view them.

Or perhaps I'm doing this wrong?

**Comment by SpencerTorres** [comment, marked as answer]
The trace search does have a limited set of columns, and if you wanted to add more you would need to switch to the SQL editor and swap the format to table. You should still be able to add filters to other columns though, even if they aren't returned in the trace search table.

Thanks for trying out the new version and for the feedback

**Comment by lucievym** [comment]
Hi, have you found out how to write the SQL query to be able to see the events in the Trace view? I cannot get past mapping them without getting an error that they are not the same data type.
